powersimdata/data_access/data_access.py

```diff
@@ -279,101 +184,59 @@ def get_profile_version(self, grid_model, kind):
-        :return: (*bool*) -- whether the file exists
-        """
-        return os.path.exists(filepath)
-
 
 class SSHDataAccess(DataAccess):
     """Interface to a remote data store, accessed via SSH."""
 
     _last_attempt = 0
 
-    def __init__(self, root=None, backup_root=None):
+    def __init__(self, root=None):
         """Constructor"""
-        super().__init__(root, backup_root)
-        self._ssh = None
+        super().__init__(root)
+        self._fs = None
         self._retry_after = 5
         self.local_root = server_setup.LOCAL_DIR
+        self.local_fs = fs2.open_fs(self.local_root)
         self.description = "server"
-        self.join = posixpath.join
 
     @property
-    def ssh(self):
-        """Get or create the ssh connection object, with attempts rate limited.
+    def fs(self):
+        """Get or create the filesystem object, with attempts rate limited.
 
         :raises IOError: if connection failed or still within retry window
-        :return: (*paramiko.SSHClient*) -- the client instance
+        :return: (*powersimdata.data_access.ssh_fs.WrapSSHFS) -- filesystem instance
         """
-        should_attempt = time.time() - SSHDataAccess._last_attempt > self._retry_after
-
-        if self._ssh is None:
+        if self._fs is None:
+            should_attempt = (
+                time.time() - SSHDataAccess._last_attempt > self._retry_after
+            )
             if should_attempt:
                 try:
-                    self._setup_server_connection()
-                    return self._ssh
+                    self._fs = get_ssh_fs(self.root)
+                    return self._fs
                 except:  # noqa
                     SSHDataAccess._last_attempt = time.time()
             msg = f"Could not connect to server, will try again after {self._retry_after} seconds"
             raise IOError(msg)
 
-        return self._ssh
-
-    def _setup_server_connection(self):
-        """This function setup the connection to the server."""
-        client = paramiko.SSHClient()
-        client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
-        try:
-            client.load_system_host_keys()
-        except IOError:
-            print("Could not find ssh host keys.")
-            ssh_known_hosts = input("Provide ssh known_hosts key file =")
-            while True:
-                try:
-                    client.load_system_host_keys(str(ssh_known_hosts))
-                    break
-                except IOError:
-                    print("Cannot read file, try again")
-                    ssh_known_hosts = input("Provide ssh known_hosts key file =")
-
-        server_user = server_setup.get_server_user()
-        client.connect(
-            server_setup.SERVER_ADDRESS,
-            username=server_user,
-            port=server_setup.SERVER_SSH_PORT,
-            timeout=10,
-        )
-
-        self._ssh = client
+        return self._fs
```

Comment:
Do we need to return the attribute?

Reply:
You mean `self._fs`? That is part of the `@property` defined above, so when we call `self.fs` it will create the filesystem if needed and return it.

Reply:
Ok. Not super familiar with `@property`